File: day01Scala/src/main/scala/com/sjyttkl/chapter01/TestScala.scala
package com.sjyttkl.chapter01

/**
  * Create with: com.sjyttkl.chapter01 
  * author: songdongdong  
  * E-mail: songdongdong@jd.com 
  * date: 2019/2/17 22:20 
  * version: 1.0
  * description:  
  */
//只要以后看到有 object TestScala ，你应该有这样一个认识
//1. object TestScala 对应的是一个  TestScala$的一个静态对象 MODULE$
//2. 在我们的程序中，是一个单例
object TestScala {
  def main(args:Array[String]) :Unit={
    println("hello , sclaa ,idea..")
  }
}

File: src/main/scala/com/sjyttkl/chapter01/test/TestScala.java
package com.sjyttkl.chapter01.test;

//import scala.Predef;

/**
 * Create with: com.sjyttkl.chapter01.test
 * author: songdongdong
 * E-mail: songdongdong@jd.com
 * date: 2019/2/17 22:25
 * version: 1.0
 * description:
 */
//我们可以理解scala 在运行时，做了一个包装.
public class TestScala {
    public static void main(String[] paramArrayOfString)
    {
        TestScala$.MODULE$.main(paramArrayOfString);
    }
}

 final class TestScala$
{
    public static final TestScala$ MODULE$;

    static
    {
//        new ();
        MODULE$ = new TestScala$();
    }

    public void main(String[] args)
    {
//        Predef..MODULE$.println("hello , sclaa ,idea..");
        System.out.println("hello , sclaa ,idea..");
    }
    //private TestScala$() { MODULE$ = this; }

}
